sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConsumerSettings.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation;
+
+import com.azure.core.amqp.AmqpConnection;
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.implementation.handler.DeliverySettleMode;
+import com.azure.core.amqp.implementation.handler.ReceiveLinkHandler;
+import com.azure.core.amqp.implementation.handler.ReceiveLinkHandler2;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.apache.qpid.proton.engine.Receiver;
+
+import java.util.Objects;
+
+// temporary type to support v1 and v2 side by side.
+public final class ConsumerSettings {
+    private final boolean isV2;
+    private final DeliverySettleMode settleMode;
+    private final boolean includeDeliveryTagInMessage;
+
+    public ConsumerSettings() {
+        this.isV2 = false;
+        this.settleMode = null;
+        this.includeDeliveryTagInMessage = false;
+    }
+
+    public ConsumerSettings(DeliverySettleMode settlingMode, boolean includeDeliveryTagInMessage) {
+        this.isV2 = true;
+        this.settleMode = Objects.requireNonNull(settlingMode);
+        this.includeDeliveryTagInMessage = includeDeliveryTagInMessage;
+    }
+
+    AmqpReceiveLink createConsumer(AmqpConnection amqpConnection, String linkName, String entityPath, Receiver receiver,
+        TokenManager tokenManager, ReactorProvider reactorProvider,
+        ReactorHandlerProvider handlerProvider, AmqpLinkProvider linkProvider, AmqpRetryOptions retryOptions) {
+        final String connectionId = amqpConnection.getId();
+        final String hostname = amqpConnection.getFullyQualifiedNamespace();
+        final AmqpMetricsProvider metricsProvider = handlerProvider.getMetricProvider(amqpConnection.getFullyQualifiedNamespace(), entityPath);
+        if (this.isV2) {
+            final ReceiveLinkHandler2 handler = handlerProvider.createReceiveLinkHandler2(connectionId, hostname, linkName, entityPath,
+                settleMode, includeDeliveryTagInMessage, reactorProvider.getReactorDispatcher(), retryOptions);
+            BaseHandler.setHandler(receiver, handler);
+            receiver.open();
```

Comment:
Great question, so the idea is - once we're entirely on V2, this file (ConsumerSettings) will be deleted, and these 4 instructions will be moved [here](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java#L611-L620). Today the ReactorSession has the responsibility to "create", "set all properties (~8 props)" and "open" the QPid Receiver before creating AmqpReceiveLink , so I was thinking of keeping the same pattern to reduce the diff.

I think we can check if it makes sense for the part where we create AmqpReceiveLink to take all those responsibilities. For this PR, I thought to keep new changes as close to the _main_ as possible to avoid large diff.